### ISSUE TITLE
Add some form of varargs

### ISSUE THREAD

**Issue by Frotty** [wurstscript/WurstScript#625]
As discussed and obvious due to recent PRs, not having varargs can cause a lot of boilerplate code if you want to support nice APIs.

There are several design possibilities:
```kotlin
function foo(int... args) // Java

function foo(vararg int args..) // Kotlin

function foo(params int[] args) // C#
```

The implementations could be:
- via list
- via global array
- via static generation of functions with the required amount of parameters

**Comment by peq** [comment]
For syntax I would either use `foo(int... args)` or `foo(vararg int args)`. The `vararg` syntax has the benefit, that it easier to search for people who don't know the feature yet, so I prefer this at the moment. However, I don't have a strong opinion on it.

Regarding implementation:

- I would not use a list, since that would implicitly create an object which is not good without garbage collection.
- The global array approach also adds some overhead.
- Generating specific functions adds no overhead, so i would prefer it.

Then the question is: What can you do with the varargs, if they are not an array/list?
I suggest to only support `for`-`in` loops over the varargs as these loops can simply be unrolled. 

We could also support getting the number of varargs and accessing varargs by index like an array. That could be done by copying the arguments to a local array.

**Comment by Frotty** [comment]
implemented by #644
